crates/core/component/dex/src/component/tests.rs

```diff
@@ -257,6 +258,59 @@ async fn single_limit_order() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+/// Builds a simple order book with a two orders, fills against them both,
+/// and checks that one of the orders is auto-closed.
+async fn check_close_on_fill() -> anyhow::Result<()> {
+    let storage = TempStorage::new().await?.apply_minimal_genesis().await?;
+    let mut state = Arc::new(StateDelta::new(storage.latest_snapshot()));
+    let mut state_tx = state.try_begin_transaction().unwrap();
+
+    let gm = asset::Cache::with_known_assets().get_unit("gm").unwrap();
+
+    let mut position_1 = SellOrder::parse_str("100gm@1gn")?.into_position(OsRng);
+    position_1.close_on_fill = true;
+    let position_2 = SellOrder::parse_str("100gm@1.1gn")?.into_position(OsRng);
```

Comment:
this is much nicer for writing tests than many of the existing methods (which were made prior to this api)